src/main/webapp/dev/js/common/instructorEnroll.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * Retrieves updated column header order and generates a header string.
+ *
+ * Example: Changes this array ['Section', 'Team', 'Name', 'Email', 'Comments']
+ * into a string = "Section|Team|Name|Email|Comments\n"
+ *
+ * @param handsontableColHeader
+ * @returns {string} updated header string
+ */
+function getUpdatedHeaderString(handsontableColHeader) {
+    const colHeaders = handsontableColHeader.join('|');
+    return colHeaders.concat('\n');
+}
+
+/**
+ * Retrieves user data rows rows in the spreadsheet interface and transforms it into a string.
+ *
+ * Null value from cell is changed to empty string after .join(). Filters empty rows in the process.
+ *
+ * Example:
+ * 2 by 5 spreadsheetData (before)
+ * ['TestSection1', 'Team1', 'null', 'test1@xample.com', 'test1comments']
+ * ['TestSection2', null, 'TestName2', 'test2@example.com', null]
+ *
+ * 2 by 5 spreadsheetData (after)
+ * "TestSection1|Team1||test1@xample.com|test1comments\n
+ *  TestSection2||TestName2|test2@example.com|\n"
+ *
+ * @param spreadsheetData
+ * @returns {string} user data rows
+ */
+function getUserDataRows(spreadsheetData) {
+    // needs to check for '' as an initial empty row with null values will be converted to e.g. "||||" after .map
+    return spreadsheetData.filter(row => (!row.every(cell => cell === null || cell === '')))
+            .map(row => row.join('|'))
+            .join('\n');
+}
+
+/**
+ * Pushes data from spreadsheetDataRows into an array.
+ * Facilitates the function loadData for the Handsontable instance.
+ *
+ * @param spreadsheetDataRows
+ * @returns {Array} updated data
+ */
+function getUpdatedData(spreadsheetDataRows) {
+    return spreadsheetDataRows.map(row => row.split('|'));
+}
+
+/**
+ * Transforms the first uppercase letter of a string into a lowercase letter.
+ * @param string
+ * @returns {string} Handsontable column header in all lowercase letters
+ */
+function unCapitalizeFirstLetter(string) {
+    return string.charAt(0).toLowerCase() + string.slice(1);
+}
+
+/**
+ * Converts returned AJAX data to a suitable format required by Handsontable.
+ * @param studentsData
+ * @param handsontableColHeader
+ * @returns {Array} student data
+ */
+function ajaxDataToHandsontableData(studentsData, handsontableColHeader) {
+    const headers = handsontableColHeader.map(unCapitalizeFirstLetter);
+    return studentsData.map(student => (headers.map(
+            header => student[header])));
+}
+
+/**
+ * Toggle the chevron image depending on the user's action.
+ * @param panelCollapse
+ * @param toggleChevron
+ */
+function toggleChevronImage(panelCollapse, toggleChevron) {
+    if ($(panelCollapse).attr('class').indexOf('checked') === -1) {
+        $(toggleChevron).addClass('glyphicon-chevron-down').removeClass('glyphicon-chevron-up');
+    } else {
+        $(toggleChevron).addClass('glyphicon-chevron-up').removeClass('glyphicon-chevron-down');
+    }
+}
+
+/**
+ * Shows the "Existing students" panel with the spreadsheet interface.
```

Comment:
existing panel only or any panel? The description and the name doesn't match.

Reply:
This has already been updated in the later commits.

Reply:
I think I missed out on that commit. For the next review I should probably just look at the final change that this PR does instead of reading commit by commit. Can you help by telling me which HTMLs are generated by GodMod so that I can skip them?

Reply:
All `.html` files in the PR are updated through GodMode, I have not edited any HTML files manually. I should probably stick to my older way of naming similar type of commits i.e. `Updated HTML files through GodMode` to prevent confusion.